Foundation/FileManager.swift

```diff
@@ -1391,15 +1794,23 @@ open class FileManager : NSObject {
         }
         return statInfo
     }
+#endif
 
     internal func _permissionsOfItem(atPath path: String) throws -> Int {
+#if os(Windows)
```

Comment:
I’d rather have a `@availability(Windows, unavailable…)` here.

Reply:
This is possible to port, but, I didn't write the mapping.

Reply:
This is just a shortcut function I added to avoid calling `attributesOfItem(atPath:)` which can be quite heavy if the only thing needed is the file permissions and is really just a shortcut version of `attributesOfItem(atPath:)[.posixPermissions].intValue`

I notice that you dont support `.posixPermissions` at the moment, I guess you could always just return the permissions as somthing like:

```
var permissions: mode_t = isReadableFile(atPath: path) ? 0o400 : 0
permissions |= isWritableFile(atPath: path) ? 0o200 : 0
```

Reply:
I'd rather not introduce any new potential crash paths as much as possible. If you don't support POSIX permissions yet, just don't return them.

Reply:
I think that crash paths are preferable to silent failures.  An explicit crash makes it easier to track down the issues once we get stuff running :-).

Reply:
_sigh_ okay, since all `NSUnimplemented`'s are `#if os(Windows)` and so _you_ have to deal with them. Just remember that any crash point you add is going to be a landmine in the codebase for a long while.

Reply:
Yeah, that's why I'm trying to keep that to a minimum and trying to implement as much as possible.  I've also annotated the functions as deprecated to try to catch them at build time if possible.